mlir/lib/Dialect/Rock/IR/RockDialect.cpp

```diff
@@ -2014,6 +2015,47 @@ LogicalResult AttentionOp::verify() {
   return success();
 }
 
+//===-----------------------------------------------------===//
+// AttentionPerfConfig Attr
+//===-----------------------------------------------------===//
+
+AttnPerfConfigAttr AttnPerfConfigAttr::get(StringAttr perfConfigStrAttr) {
+  // Here a conventional c++ string split is being
+  // done because MLIR lacks parseSourceString() method
+  // to parse Attributes and its only there for Ops.
+  std::string perfConfigStr = perfConfigStrAttr.str();
```

Comment:
You should be able to do all this on `StringRef` to avoid the copies etc.?

Reply:
Similarly, this parser should probably take `StringRef`

Reply:
I like the parser interface to take StringAttr -- and it also avoid copies but carries the context.
What I d do here is i ll use strref() insterad str() ?

Reply:
StringRef doesn't copy - it's a reference.

Reply:
I know... 

What Im saying is I ll take strref of storage of StringAttr which is provided by StringAttr::strref().
and I like to keep the builder StringAttr -> AttnPerfConfigAttr.

Reply:
moved to StringRef based parsing.